### ISSUE TITLE
Electric Drive Enumeration fault debugging?

### ISSUE THREAD

**Issue by DeanKoyanagi** [directedmachines/customer-support#19]
I had finally replaced the broken wheel hubs with the welded steel hubs/adapters which required some grinding down of the damaged areas on the drive shaft from the previous aluminum hubs breaking. I drove the LCR in manual mode for a few days checking frequently for bolts or set screws loosening up. They all seemed to be holding with a coating of Locktite.

While walking the LCR in the grass along the driveway, I drove over a stone I probably had plowed out of our gravel driveway. It was maybe 2" thick?  Nothing unusual for our rocky soils and something our LCR usually cruises over...) This time, the LCR came to immediate hard stop, and I Iost Wifi connection again. (It has done that multiple times recently when I stop the LRC, and then I have to reconnect to the Wifi to continue). 

I again checked for loose USB connectors and ensured all the antenna connections and wires are isolated from the frame and the wires show no sign of abrasion damage. I can connect to the WifI, but I get the Electric Drive Enumeration fault error. I found the Chassis Short Mitigation document (https://github.com/directedmachines/customer-support/wiki/ChassisShortMitigation.pdf) but I did not get any red light as described to signify a short. Any other suggestions for debugging an Electric Drive Enumeration fault? If not related, then we can worry later about why the WIfi keeps disconnecting.

**Comment by georgechrysanthakopoulos** [comment]
loss of wifi is not a good sign: it means the CPU is rebooting. that can happen if there is some kind of wiring fault on the logic side.

i can connect remotely and see if I can discover anything.

**Comment by georgechrysanthakopoulos** [comment]
@DeanKoyanagi robot currently not connected to cell network, so I can note remotely connect. I have seen it connect, plus tons of telemetry, so maybe there is a better spot for reception?
<img width="380" alt="Screen Shot 2021-08-10 at 10 21 34 AM" src="https://user-images.githubusercontent.com/978440/128905807-caa660b0-07e7-45e8-83b6-46260204e8d8.png">

**Comment by DeanKoyanagi** [comment]
Sorry Georgios,

I had it under a tarp for the 60mpg gusts and severe thunderstorm that
passed by us the other day. It may have been too much for the puck (which
seems to have less Wifi range than the old antenna, which might also be the
case with the LTE? Lights were on the LTE when I just checked.

Dean

**Comment by georgechrysanthakopoulos** [comment]
@DeanKoyanagi the CPU and our software is alive and is connecting, but its intermittent. LTE reception not great. here is what I suggest to clear the electrical fault:
1. shutdown our software, using the UI (gear icon, power icon, "shutdown"
2. remove the pink fuse for the pi/usb HUB
3. toggle off the USB 7 port hub (use its switch to turn it off)
4. Turn off the main battery disconnect switch
5. wait 1 minute for capacitors to drain
6. Turn battery disconnect switch back on
7. turn hub back on
8. plug in pink fuse
9. wait about 2 minutes until PI OS restarts, WIFI broadcasts, UI is available. Fault should be cleared.

i would still check ALL motor connections, from motor terminals, to motor controller motor + and - terminals plus battery +/- terminals to motor controllers.

**Comment by georgechrysanthakopoulos** [comment]
I was able to remotely connect. Something is damaged with your USB cables. We sent you brand new cables for everything, so not sure how this happened. I also verified that none of the 300A motor controllers are enumerating. We just see the GPS and likely the 50AMC controller.

The cameras do not appear connected either but the USB bus is in bad shape right, so can not tell much else
Last login: Wed Aug 11 16:33:11 2021
pi@LCR24ZS0-6462231508f3a1c37:~ $ ls -la /dev/serial/by-id
total 0
drwxr-xr-x 2 root root 80 Aug 11 16:33 .
drwxr-xr-x 4 root root 80 Aug 11 16:33 ..
lrwxrwxrwx 1 root root 13 Aug 11 16:33 usb-STMicroelectronics_STM32_Virtual_ComPort_207536535256-if00 -> ../../ttyACM0
lrwxrwxrwx 1 root root 13 Aug 11 16:33 usb-u-blox_AG_-_www.u-blox.com_u-blox_GNSS_receiver-if00 -> ../../ttyACM1
pi@LCR24ZS0-6462231508f3a1c37:~ $ dmesg
[ 1452.909066] usb usb2-port2: Cannot enable. Maybe the USB cable is bad?
[ 1457.286663] usb usb2-port2: Cannot enable. Maybe the USB cable is bad?
[ 1457.287023] usb usb2-port2: attempt power cycle
[ 1461.974153] usb usb2-port2: Cannot enable. Maybe the USB cable is bad?
[ 1466.311865] usb usb2-port2: Cannot enable. Maybe the USB cable is bad?
[ 1466.312226] usb usb2-port2: unable to enumerate USB device
[ 1470.639599] usb usb2-port2: Cannot enable. Maybe the USB cable is bad?
